### PR TITLE
dynamic: disable ONLY stops -test from auto running

### PR DESCRIPTION
**Pull request by jfoug**
Fix for #3970
Fix for #4002

### CODE REVIEW

**Comment by jfoug** [comment]
Changed the 'logic' for the disabled formats.  Now the format gets loaded.  However, we simply do not auto test.  BUT if the hash is in an input file, and we are running, then the hash will auto detect.

As always, ```--format=dynamic-all``` will test all dynamic hashes (even the disabled ones).   Also in debugging builds, all dynamic is tested (as it was before).

**Comment by jfoug** [comment]
This is actually better logic.  We simply added this, to cut down on time (especially in the CI's) which the --test was taking.  Repetitively going over these junk formats was not doing anything about finding any bugs.   BUT totally disabling the formats, i.e. where they were not working without using the --format switch really was wrong run time behavior.

File: run/dynamic_disabled.conf [hunk 1]
```diff
@@ -135,10 +137,10 @@ dynamic_213 = Y
 dynamic_214 = Y
 dynamic_215 = Y
 dynamic_216 = Y
-dynamic_217 = Y
 dynamic_218 = Y
 dynamic_221 = Y
 dynamic_222 = Y
+dynamic_217 = Y
 dynamic_223 = Y
 dynamic_224 = Y
```
Comment (magnumripper) [inline run/dynamic_disabled.conf]: Why the reordering here?

**Review by magnumripper — APPROVED** [review]
You should revert the inadvertent move of dynamic_217 line. Other than that, it looks reasonable to me.

File: run/dynamic_disabled.conf (hunk 1 above)
Comment (jfoug) [reply run/dynamic_disabled.conf]: Not sure how the swap originated, but this will get it back to original form.

**Review by solardiz — APPROVED** [review]
Looks good enough to me.

**Comment by jfoug** [comment]
I see this is committed.  But at work today, I was wondering if we should have renamed the section.  Currently it is called "[Disabled:Formats]"   I was thinking along the lines of "[Muted:Format-tests]"  

But also does this cause problems for people who have a customized john.conf file?  

It is sort of a moot point for this commit, but since we have enabled the formats (but kept the testing muted), where before the entire format was muted (since we did not load the format at startup.

I am just interested in how others feel.  john works 100% like it should now.  This is more for documenting names to match behavior.

**Comment by magnumripper** [comment]
Good point, but yes I think we'll only introduce head ache if renaming it

File: src/bench.c [hunk 2]
```diff
@@ -747,6 +747,27 @@ int benchmark_all(void)
 /* Silently skip formats for which we have no tests, unless forced */
 		if (!format->params.tests && format != fmt_list)
 			continue;
+		
+/* Format disabled in john.conf */
+		if (cfg_get_bool(SECTION_DISABLED, SUBSECTION_FORMATS,
```
Comment (magnumripper) [inline src/bench.c:752]: Very strange: Even though line 750 looks fine here, and this supposedly is the very commit that was used, it now has whitespace noise (two tabs).

**Pull request merged**